src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/SplitIntoConsecutiveIfStatementsTests.cs

```diff
@@ -0,0 +1,1324 @@
+        }
+        else if (b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task SplitNotIntoSeparateStatementsIfControlFlowContinues2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a [||]|| b)
+        {
+            if (a)
+                return;
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+            if (a)
+                return;
+        }
+        else if (b)
+        {
+            if (a)
+                return;
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task SplitNotIntoSeparateStatementsIfControlFlowContinues3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a [||]|| b)
+            while (a)
+            {
+                break;
+            }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+            while (a)
+            {
+                break;
+            }
+        else if (b)
+            while (a)
+            {
+                break;
+            }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task SplitNotIntoSeparateStatementsIfControlFlowContinues4()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        while (true)
+        {
+            if (a [||]|| b)
+                switch (a)
+                {
+                    default:
+                        break;
+                }
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        while (true)
+        {
+            if (a)
+                switch (a)
+                {
+                    default:
+                        break;
+                }
+            else if (b)
+                switch (a)
+                {
+                    default:
+                        break;
+                }
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task SplitNotIntoSeparateStatementsWithElseIfControlFlowQuits()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a [||]|| b)
+            return;
+        else
+            return;
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+            return;
+        else if (b)
+            return;
+        else
+            return;
+    }
+}");
+        }
+
+        [Fact]
+        public async Task SplitNotIntoSeparateStatementsAsEmbeddedStatementIfControlFlowQuits1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        while (a)
+            if (a [||]|| b)
+                return;
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        while (a)
+            if (a)
+                return;
+            else if (b)
```

Comment:
This won't split because we'd have to add braces? Are we better off just adding braces in that case?

Reply:
It definitely would be better. Although I'm wondering what the best way would be to implement that.

Should SyntaxGenerator.WithStatements support blocks and embedded statement owners in addition to declarations? That raises lots of questions like: if there's just 1 statement, would it always put it inside a block? what if we pass in 1 statement that's already a block? Plus, this PR is already huge. I would rather avoid adding new functionality to a public API on the SyntaxGenerator.

The only thing we need to do is insert a new statement, so it seems this might be best handled by the SyntaxEditor (which I'm currently using), but SyntaxEditor is completely language agnostic.

Or should we add SyntaxGenerator.ScopeBlock and then use that in the feature, even though this method would have to throw in VB? And then is it a good assumption that any language that has embedded statements will support scope blocks in order to be able to introduce more statements?